evmlib/src/external_signer.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2024 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::common::{Address, Amount, Calldata, QuoteHash, QuotePayment, U256};
+use crate::contract::data_payments::{DataPaymentsHandler, MAX_TRANSFERS_PER_TRANSACTION};
+use crate::contract::network_token::NetworkToken;
+use crate::contract::{data_payments, network_token};
+use crate::utils::http_provider;
+use crate::Network;
+use std::collections::HashMap;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Network token contract error: {0}")]
+    NetworkTokenContract(#[from] network_token::Error),
+    #[error("Chunk payments contract error: {0}")]
+    ChunkPaymentsContract(#[from] data_payments::error::Error),
+}
+
+#[allow(dead_code)]
```

Comment:
Why are these needed for `pub`?

Reply:
The Error struct? It needs the same visibility as the `pay_for_quotes_calldata` function that potentially returns it

Reply:
I meant all `#[allow(dead_code)]` attributes in those file, sorry, don't know why GitHub showed the lines before that.

Reply:
Good catch, removed them all now